drivers/clock_control/nrf_power_clock.c

```diff
@@ -37,15 +38,27 @@ struct nrf_clock_control {
 };
 
 /* Clock instance static configuration */
-struct nrf_clock_control_config {
+struct nrf_clock_control_sub_config {
 	nrf_clock_handler_t start_handler; /* Called before start */
 	nrf_clock_handler_t stop_handler; /* Called before stop */
 	nrf_clock_event_t started_evt;	/* Clock started event */
 	nrf_clock_task_t start_tsk;	/* Clock start task */
 	nrf_clock_task_t stop_tsk;	/* Clock stop task */
+	const char *name;
+};
+
+struct nrf_clock_control_data {
+	struct nrf_clock_control data[NRF_CLOCK_TYPE_COUNT];
 };
 
-static void clkstarted_handle(struct device *dev);
+struct nrf_clock_control_config {
+	struct nrf_clock_control_sub_config config[NRF_CLOCK_TYPE_COUNT];
```

Comment:
Perhaps it would be better to name this field `subsys`, to avoid references like `config->config[type]` (see [line 117](https://github.com/zephyrproject-rtos/zephyr/pull/21298/files#diff-3c5c76aad21e56267c253f5caa8c09e8R117)). Same story with `data` in `struct nrf_clock_control_data`.

Reply:
done